src/managers/ApplicationCommandManager.js

```diff
@@ -0,0 +1,201 @@
+   * @param {ApplicationCommandData[]} commands The commands
+   * @returns {Promise<Collection<Snowflake, ApplicationCommand>>}
+   */
+  async set(commands) {
+    const data = await this.commandPath.put({
+      data: commands.map(c => this.constructor.transformCommand(c)),
+    });
+    return data.reduce((coll, command) => coll.set(command.id, this.add(command)), new Collection());
+  }
+
+  /**
+   * Edits an application command.
+   * @param {ApplicationCommandResolvable} command The command to edit
+   * @param {ApplicationCommandData} data The data to update the command with
+   * @returns {Promise<ApplicationCommand>}
+   */
+  async edit(command, data) {
+    const id = this.resolveID(command);
+    if (!id) throw new TypeError('INVALID_TYPE', 'command', 'ApplicationCommandResolvable');
+
+    const raw = {};
+    if (data.name) raw.name = data.name;
+    if (data.description) raw.description = data.description;
+    if (data.options) raw.options = data.options.map(o => ApplicationCommand.transformOption(o));
+    if (data.defaultPermission) raw.default_permission = data.defaultPermission;
+
+    const patched = await this.commandPath(id).patch({ data: raw });
+    return this.add(patched);
+  }
+
+  /**
+   * Deletes an application command.
+   * @param {ApplicationCommandResolvable} command The command to delete
+   * @returns {Promise<?ApplicationCommand>}
+   */
+  async delete(command) {
+    const id = this.resolveID(command);
+    if (!id) throw new TypeError('INVALID_TYPE', 'command', 'ApplicationCommandResolvable');
+
+    await this.commandPath(id).delete();
+
+    const cached = this.cache.get(id);
+    this.cache.delete(id);
+    return cached ?? null;
+  }
+
+  /**
+   * Fetches the permissions for one or multiple commands.
+   * @param {ApplicationCommandResolvable} [command] The command to get the permissions from
+   * @returns {Promise<ApplicationCommandPermissions[]|Collection<Snowflake, ApplicationCommandPermissions[]>>}
+   */
+  async fetchPermissions(command) {
+    if (command) {
+      const id = this.resolveID(command);
+      if (!id) throw new TypeError('INVALID_TYPE', 'command', 'ApplicationCommandResolvable');
+
+      const data = await this.commandPath(id).permissions.get();
+      return data.permissions.map(o => this.constructor.transformPermissions(o, true));
+    }
+
+    const commands = await this.commandPath.permissions.get();
+    return commands.reduce(
+      (coll, data) =>
+        coll.set(
+          data.id,
+          data.permissions.map(o => this.constructor.transformPermissions(o, true)),
+        ),
+      new Collection(),
+    );
+  }
+
+  /**
+   * Sets the permissions for a command.
+   * @param {ApplicationCommandResolvable} command The command to edit the permissions for
+   * @param {ApplicationCommandPermissions[]} permissions The new permissions for the command
+   * @returns {Promise<?ApplicationCommand>}
+   */
+  async setPermissions(command, permissions) {
```

Comment:
It should also accept an array, as you can edit several application command permissions with one call.